### ISSUE TITLE
False values shown as empty items when showThumbs is false

### ISSUE THREAD

**Issue by alkafaiz** [leandrowd/react-responsive-carousel#563]
I want to have dynamic items for the carousel. the code is something like this:
 ```
return (
    <Carousel autoPlay>
      {images.length
        ? images.map((img, index) => (
            <div key={index}>
              <img alt="" src={img} />
            </div>
          ))
        : null}
    </Carousel>
  );
```

the behaviour that I encountered: 

1. when images.length is 0, it will show error of `TypeError: Cannot read property 'type' of null`
2. when I set `showThumbs` to false, the error is gone but the `null `value becomes empty item of the carousel

here is the screenshot of the rendered empty item
![image](https://user-images.githubusercontent.com/23336030/108206514-73976480-7161-11eb-97b2-d92b7a021bfe.png)

my question is how to not show empty items when it has dynamic item rendering and it has the possibility to return false value? thank you

**Comment by renancleyson-dev** [comment]
You probably are using javascript, but on typescript, I get this error:
```
Type 'Element[] | null' is not assignable to type 'ReactChild[] | (ReactChild[] & string) | (ReactChild[] & number) | (ReactChild[] & false) | (ReactChild[] & true) | (ReactChild[] & ReactElement<...>) | (ReactChild[] & ReactNodeArray) | (ReactChild[] & ReactPortal) | undefined'
```
It simply means that null isn't a valid value to `children` prop, but it shows too that an empty array and `undefined` are valid values. In fact, the carousel handles it and returns null if there's no child or `children` prop at all.

So using just this code will work out:
```javascript
{images.map((img, index) => (
    <div key={index}>
        <img alt="" src={img} />
    </div>
))}
```

**Comment by renancleyson-dev** [comment]
I tried to reproduce your issue since even if the carousel doesn't expect to receive `null` on the `children` prop, it will return `null` if the `children` prop is falsy, so it should work if you aren't using typescript.

> the behaviour that I encountered:   
>     1. when images.length is 0, it will show error of `TypeError: Cannot read property 'type' of null`
>     2. when I set `showThumbs` to false, the error is gone but the `null `value becomes empty item of the carousel

Your example really worked when I tested it on a [sandbox](https://codesandbox.io/s/awesome-cannon-ffbu1), but it didn't render the carousel when the `images` array was empty too. So it doesn't have the same behavior as the broken code.

Looking into the code related to how the thumbs render:
```typescript
const images = Children.map<React.ReactChild | undefined, React.ReactChild>(children, (item) => {
                let img: React.ReactChild | undefined = item;

                // if the item is not an image, try to find the first image in the item's children.
                if ((item as React.ReactElement<{ children: React.ReactChild[] }>).type !== 'img') {
                    img = (Children.toArray((item as React.ReactElement).props.children) as React.ReactChild[]).find(
                        (children) => (children as React.ReactElement).type === 'img'
                    );
                }

                if (!img) {
                    return undefined;
                }

                return img;
            });
}
```
The error occurs when there's a child in the `children` array that is `null`, which will try to access the `type` property and show the message `TypeError: Cannot read property 'type' of null`. So it's not the `children` prop with a `null` value that breaks the code, but it's a `children` prop array with some `null` values.

If my explanation is correct, you can check if you are passing an array that can have `null` values as children and try to remove it somehow.

**Comment by alkafaiz** [comment]
> If my explanation is correct, you can check if you are passing an array that can have `null` values as children and try to remove it somehow.

Well, thanks for the explanation. Yes, I use Javascript, not Typescript, and you are right to just map the array without wrapping it inside a conditional. If the array is empty, it would not display an empty carousel item. 

However, does this mean that we should not put conditional statement which has possible result of falsy value inside the carousel. In my use case, I need to make something like this:

 ```
return (
    <Carousel>
      {images.map((img, index) => (
        <div key={index}>
          <img alt="" src={img} />
        </div>
      ))}
      {!isMax && <div>Add more</div>}
    </Carousel>
  );
```

This snippet will always show an extra empty carousel item if the `isMax` is `true`. I have tried to do a typical if statement with the false value of `null`, `undefined`, empty string, and an empty fragment `<></>` but all still return empty carousel item.

I've managed to work around this tho, by including the item that I wanted to display conditionally into the array. I'll be nicer if a falsy value can be just ignored by the carousel.
